src/NanopassSharp/Patterns/BinaryPattern.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+
+namespace NanopassSharp.Patterns;
+
+/// <summary>
+/// A pattern which applies a binary operation to two other patterns.
+/// </summary>
+public sealed class BinaryPattern : ITransformationPattern
```

Comment:
I feel very similarly of these to transformations, they mostly factor out a few lambdas and it cost you a class and a builder function.